src/Tools/DynamoShapeManager/Preloader.cs

```diff
@@ -139,7 +139,11 @@ public Preloader(string rootFolder, IEnumerable<Version> versions)
             shapeManagerPath = string.Empty; // Folder that contains ASM binaries.
             version = Utilities.GetInstalledAsmVersion2(versionList, ref shapeManagerPath, rootFolder);
 
-            var libGFolderName = string.Format("libg_{0}_{1}_{2}", version.Major, version.Minor, version.Build);
+            var libGFolderName = "";
+            if (version != null)
+            {
+                libGFolderName = string.Format("libg_{0}_{1}_{2}", version.Major, version.Minor, version.Build);
```

Comment:
If version is null, then libGFolderName won't be correct (empty).  And then preloaderLocation/geometryFactoryPath will not be pointing to any dlls, but they will not be empty either.  Worse yet, geometryFactoryPath will not even be a correct path.
I think it would be better that preloaderLocation/geometryFactoryPath be empty so that you can later check if they're not empty.

```suggestion
            if (version == null)
            {
                return;
            }
```
Just returning will/should keep preloaderLocation/geometryFactoryPath not set/empty.

Reply:
Thanks, this is a good point. I'm not sure what the consequences of leaving these paths empty might be. The intent of this change is just to fix the crash, so preserving the earlier behavior of setting up valid but nonexistent `libg_0` path seems the safest route. Additional work in this area would definitely be warranted.